Add .md files on corpora

idalr opened the pull request:
In this PR, we added the corpora documentation on our six corpora: AAE2, AbsTRCT, ArgMicro, CDCP, SciArg, and SciDTB-argmin.

In each markdown file, we provide the description of the corpus, general information, external resources, example, its components and relations, data structure, and collected statistics (see [metrics](https://github.com/ArneBinder/pie-document-level/tree/main/configs/metric)).

In `CONCEPTS.md`, we also provide some concept explanations, regarding argument mining, annotation, components, etc.

idalr committed:
Add files via upload

ArneBinder commented on AM corpora/aae2.md:
```diff
@@ -0,0 +1,140 @@
+		- attack; Attack: 219
+	- Segment-based spans: No
+- Table 2: Statistics *at paragraph-level*:
+	- type: tree
```
What does the `#` or `%` mean? Is it just to mark indentation? If so, please just indent these lines accordingly.

idalr replied:
Fixed in 54927d8 for clarity. Originally # refers to count and % refers to percent.

ArneBinder commented on AM corpora/aae2.md:
```diff
@@ -0,0 +1,140 @@
+Summary from Mario et al.:
+- AAEC (Stab and Gurevych, 2017) contains student essays. There are two types of data: essay-level and paragraph-level (Eger et al., 2017). A stance for a controversial theme is expressed by a MajorClaim component as well as Claim components, and Premise components justify or refute the Claims. Attack and Support labels are defined as relations. The span covers a statement, which an stand in isol ...
```
- quotation marks should be written just as quotes: `"shell"` instead of `´´shell´´`. I think some other quotation marks are missing (e.g. "on the other hand")?
- if there are concepts in the main texts that are no real words (e.g. `MainClaim`) or, in general, have a certain meaning defined here (e.g. `Premise`), put them in tics to mark them as code (see here: https://www.markdownguide.org/basic-syntax/#code)
- please add links to the references (see e.g. this [markdown how-to](https://www.markdownguide.org/basic-syntax/#links)) either directly in the text or as a list after the text

ArneBinder replied:
The author is called "Morio", not "Mario".

idalr replied:
fixed in 54927d8

ArneBinder commented on AM corpora/aae2.md:
```diff
@@ -0,0 +1,140 @@
+	- Relation
+		- support; Support: 3613
+		- attack; Attack: 219
+	- Segment-based spans: No
```
can you put this in a markdown-table? 
 - this describes the syntax: https://www.codecademy.com/resources/docs/markdown/tables
 - but there are also table generators, e.g. https://www.tablesgenerator.com/markdown_tables

ArneBinder replied:
Can you also add short definitions (maybe approx. one sentence) for the individual types? I'm not 100% sure if this should be in a table or just in an indented bullet list, decide for yourself what looks better.

idalr replied:
fixed in 54927d8. Arranged in a table, with explanation in bullet list below

ArneBinder commented on AM corpora/aae2.md:
```diff
@@ -0,0 +1,140 @@
+		- id: list
+		- type: list
+		- target: list
+		- note: list
```
This is not very well readable. To improve, you could:
 - remove everything that comes from the [BRAT standoff format](https://brat.nlplab.org/standoff.html), just reference their documentation
 - use the document type (e.g. see `BratDocument` or `BratDocumentWithMergedSpans` from [here](https://huggingface.co/datasets/pie/brat/blob/main/brat.py)) instead of the dictionary representation to describe the format
 - do not list examples per field, but give one single example at the end
 - in general: keep it short as best as possible, try to reference external information if available

ArneBinder replied:
Could you also add a figure that shows a data example, maybe take a screenshot from the dataset paper?

idalr replied:
improved in cc8debe.
- add reference to BRAT standoff format
- not sure if what I took from `BratDocument` in `brat.py` makes sense. If not, please let me know
- create one data sample in a table

ArneBinder replied:
Looks good in general, regarding the "Data fields" section: You could rename that to e.g. "Document Structure" or "Data Structure" and separate the fields into "data fields" (context, filename) and annotation layers (spans, relations, ...). Then, you can name the python type + default, if available, for all data types and the annotation type + target for all annotation layers.

idalr replied:
improved in 6c597370b5319028a8a5639b0df123930e164aae

ArneBinder commented on AM corpora/aae2.md:
```diff
@@ -0,0 +1,140 @@
+		- note: list
+
+Collected statistics
+- via evaluate_documents
```
Can you add the exact command you used to get this? So that we can re-calculate, in the case that anything changes / is fixed in the future.
```suggestion
- via evaluate_documents: `python src/evaluate_documents.py ...`
```

idalr replied:
fixed in https://github.com/ArneBinder/pie-document-level/commit/54927d84d728be98b880aa5da4e755b6be0aba4a

ArneBinder commented on AM corpora/aae2.md:
```diff
@@ -0,0 +1,140 @@
+		| std  |  311.772 |
+		| min  | 1329     |
+		| max  | 2690     |
+
```
Please also replace with markdown tables. I improved the logging a bit, if you pull the latest changes, the statistics are also written into a markdown file (the exact location is written on the console) whose content you can copy.

ArneBinder replied:
Also compare with Table 1 of Morio et al. (2022). Are there any differences? if so, 
 - first, please list them. 
 - Then, try to explain them (we can also look into this together).

idalr replied:
improved in cc8debe and 2026e10:
- arranged in md tables
- included stats from the whole dataset (402 docs) and when separated as train, test, valid datasets

further work:
- compare and list out differences between these stats and in the original paper (Stab&Gurevych '17), and explain the difference
- add new statistics: `relation_argument_distances` and `relation_argument_tokens_distances`

ArneBinder replied:
As discussed, I think it is enough to have the statistics for the splits as they are in the original dataset. Can you use the commands that we tried in our meeting to re-create the tables in such a way?

idalr replied:
Regarding statistics in datasets:
- the statistics comparison section in `aae2.md`. Numbers are approximately aligned but not completely.
- I haven't proceed with other datasets, but I saw discrepancies between 1) statistics in the original paper, 2) statistics reported by Morio et al., and also 3) what we evaluate ourselves via commands
- I couldn't pull out the statistics of the whole dataset from `abstrct` and `cdcp` with `+dataset.create_validation_split._enabled_=false +dataset.create_test_split._enabled_=false`. I assume the structure of dataset and `<dataset>_prepared.yaml` are a bit different. I tried what I though would work, but didn't manage.

ArneBinder commented on AM corpora/abstrct.md:
```diff
@@ -0,0 +1,137 @@
+Summary from Mario et al.:
+- AbstRCT (Mayer et al., 2020) includes abstracts of randomized controlled trials (RCTs) from the MEDLINE database.3 MajorClaim, Claim, and Evidence components, along with Support, Attack, and Partial-attack relations, are annotated for various diseases (e.g., neoplasm, glaucoma, hepatitis, diabetes, and hypertension). Similar to AAEC, this corpus discriminates argumentative text units from non- ...
```
Do the components include or exclude "shell language" (see `corpora/aae2.md`)?

idalr replied:
You're right. In `abstrct`, they left out the "shell" language

ArneBinder commented on AM corpora/argmicro.md:
```diff
@@ -0,0 +1,112 @@
+Summary from Mario et al.:
+- aka MTC (Peldszus and Stede, 2016) is based on Freeman’s theory of the macro-structure of arguments (Freeman, 2011) and incorporates the ideas of Toulmin (Toulmin, 2003) into diagramming techniques. MTC introduces dialectical exchange between Pro (proponent) and Opp (opponent) components. Relations include Add, Exa (example), Reb (rebut), Sup (support), and Und (undercut). According to preceden ...
```
What does it mean to "pre-process the Add relations similarly to Kuribayashi et al. (2019)"?

idalr replied:
I left this part out in the recent commit (d2b27a4193a1c5448ad10f1666483e7382b34a7f), as this was the part on how Morio et al. were processing `argmicro` for their own experiment

ArneBinder commented on AM corpora/sciarg.md:
```diff
@@ -0,0 +1,115 @@
+Summary:
+- enriching Dr. Inventor corpus, consisting of 40 publications from computer graphics, with argumentative information
```
This is a bit short, can you extend this?

idalr replied:
extended in recent commit a59c87a6eb3d23fdf51be6a0bcc5c811f622d634

ArneBinder commented on AM corpora/terms explained.md:
```diff
@@ -0,0 +1,52 @@
+In Mario et al. 2022
+- tree vs graph:
+	- tree: structure with one root node
+	- graph: structure where nodes interlink in different directions, allowing transitive relations, reentracies, etc.
```
Just as a side note: Would be interesting to know if the argumentative graphs are DAGs, directed acyclic graphs, or if they are allowed to contain cycles. Would it be a problem if they contain cycles?

idalr replied:
At least in `sciarg` it was explicitly stated that they were DAGs. But, nothing is indicated in `cdcp` and `abstrct` - where they were also graphs. None of the papers discussed argumentative cyclic graphs so far.

However, Stab and Gurevych (2017) wrote something like graphs indicated complexity and divergence in arguments (pp.626-627)

ArneBinder replied:
Ok, thanks for the investigation.

ArneBinder commented on AM corpora/terms explained.md:
```diff
@@ -0,0 +1,52 @@
+- tree vs graph:
+	- tree: structure with one root node
+	- graph: structure where nodes interlink in different directions, allowing transitive relations, reentracies, etc.
+- relation density: average value of $N_r/(N_c − 1)$
```
oh, cool, I did not know that it is possible to have inline latex-math!

ArneBinder commented on AM corpora/terms explained.md:
```diff
@@ -0,0 +1,52 @@
+		- "if X provides rationale for Y"
+	- Evidence
+		- "if it proves whether proposition Y is true or not"
+		- limited to TESTIMONY or REFERENCE
```
Everything that is specific to the dataset should just be mentioned in the respective markdown file (e.g. `argmicro.md`)

idalr replied:
fix in recent commit 6c597370b5319028a8a5639b0df123930e164aae

ArneBinder commented on AM corpora/terms explained.md:
```diff
@@ -0,0 +1,52 @@
+In Mario et al. 2022
```
Rename the file to `CONCEPTS.md` (in capital letters to differenciate from actual dataset discriptions).

ArneBinder replied:
Also, link to this file from the dataset files where it is helpful to know about the concepts.

idalr replied:
in 6c597370b5319028a8a5639b0df123930e164aae:
- renamed the file
- linked concepts from \<dataset>.md to this file

ArneBinder reviewed (changes requested):
Thanks a lot for this extensive write up! There is already a lot of good content in it. Still, I added a lot of comments, especially to the first file (most of them apply also for the remaining ones, so please handle them everywhere where you think it makes sense). Please don't be overwhelmed by that, it looks quite good for the first version of this PR.

Some additional general notes:
 - add a small description to this PR: what it does and why (if you are not sure about it, you can write an initial proposal and ping me with e.g. `@ArneBinder did you mean this?` at the end.)
 - execute `pre-commit run -a` before pushing any files to github. The command should finish without complains. This also runs on GitHub via continuous integration (CI) workflows. Have a look at the small red cross at the bottom of the PR, that should become a green check mark when you have pushed the fixed files.
 - make the folder `AM corpora` a subfolder of another one (e.g. `AM documentation/corpora` or `AM knowledge base/corpora`). We will extend on that later on, e.g. with a `models` or `approaches` folder.
 - add sub-headings: see https://www.markdownguide.org/basic-syntax/#headings
 - add some more features that we can later use to compile an overview table, e.g.
 	- `text domain`: scientific, ...
   	- `text granularity` (maybe chose a better name): Is this full text, just abstracts, or paragraphs? An indicator is what components from which **parts of the text** are connected by argumentative relations in the dataset.
 - see the dataset PRs for further inspiration for what information can be added: https://github.com/ArneBinder/pie-document-level/pulls?q=is%3Apr+is%3Aclosed+label%3A%22new+dataset%22

If anything I wrote is ambiguous, does not make any sense to you, etc, please let me know!

idalr committed:
partially improved aae2.md readibility

idalr committed:
tested commit and push, changed folder name

idalr committed:
edited aae2.md and concepts.md

idalr committed:
add .obsidian to gitignore

idalr committed:
improved statistics tables in aae2.md

idalr committed:
improved statistics tables in aae2.md

idalr committed:
Delete sample_aae2.csv

idalr committed:
updated files: aae2.md and abstrct.md

idalr committed:
improved readability - all files

idalr committed:
minor changes

idalr committed:
minor changes

idalr committed:
minor changes

idalr committed:
minor changes

ArneBinder committed:
execute pre-commit

ArneBinder committed:
exclude "ist" from spell checking

ArneBinder commented on AM_documentation/corpora/CONCEPTS.md:
```diff
@@ -0,0 +1,57 @@
+#### segment-based span
+
+- spans that also contain non-argumentative component parts, usually contains a clause or sentence. (p. 642) For example, *on the other hand*, *we propose*, *because*, *this suggests*, etc.
+- on the other hand, "shell" language, or minimal unit span, has minimum boundaries and discriminates argumentative component parts from non-argumentative parts in the clause/sentences.
```
This needs some clarification (this is very bad written in Morio et al.). As I understand it, `segment-based spans` means that the text is *segmented* into spans that cover the whole text (i.e. when merging the spans it reproduces the original text). As a result, this also contains `shell` language (that are mostly rhetorical discourse markers such as "therefore", "on the other hand", etc) or other non-argumentative content. In contrast, `non-segment-based spans` do not need to cover the whole input text and should only consist of argumentative content (in the context of ADUs/AM).

Does this make sense or do you see any inconsistencies?

idalr replied:
As we spoke last week, in commit de8352b, I added `segment-based span`, `non-ADUs`, and `shell language` in every corpus file; also defined `segment-based spans` and `shell language`  in the CONCEPTS.md file.

idalr replied:
I couldn't find the way to comment on the top comment...so, please allow me to add comments here:
In the recent commit (de8352b), I also added these new points:
- "pre-defined splits"
- "text-based or token-based"
- "contains ADUs"
- "contains shell language"
- "pre-processed by Morio et al." (only if needed)
- "example" (screenshot of example)
- column of percentage of components and relations, next to Morio et al.'s statistics
- `count_relation_labels` under "Collected Statistics" section - and also displayed statistics according to pre-defined split (if exists)

ArneBinder replied:
fyi regarding comments to the top comment: you can just add a comment to the very bottom of the PR (below the checks and merging section)

ArneBinder commented on AM_documentation/corpora/CONCEPTS.md:
```diff
@@ -0,0 +1,57 @@
+#### Joint
+
+a.k.a linked argument (Freeman's theory, [1991](<(https://api.pageplace.de/preview/DT0400.9783110875843_A19822678/preview-9783110875843_A19822678.pdf)>)  [2011](https://link.springer.com/book/10.1007/978-94-007-0357-5))
+"a joint of two or more adjacent elementary discourse units When two adjacent EDUs are argumentatively relevant only when taken together, these EDUs are first connected with one joint EDU node by segmentation edges and then this joint node is connected to a corresponding ADU."
```
Is this really about joint "EDUs", not "ADUs"? Compare example "en/micro_b001" from the argmicro corpus. Here it looks like multiple ADUs participate jointly in a support relation. 

Is `Joint` a "general" concept at all? Or specific to the argmicro corpus?

Or is this the same as a `linked argument` (see section "convergent argument" above)?

idalr replied:
corrected in de8352b.
`JOINT` is indeed the concept in ArgMicro although this is similar to `parts_of_same` in SciArg. This is NOT a linked argument, as I thought it was. `JOINT` is in the EDUs level - supposedly when a text is separated by punctuation. The given example in the guideline is EDU1 = [The building is contaminated by asbestos.] and EDU2 = [Through and through!]

ArneBinder commented on AM_documentation/corpora/aae2.md:
```diff
@@ -0,0 +1,235 @@
+
+### Summary from [Morio et al. (2022)](https://aclanthology.org/2022.tacl-1.37.pdf):
+
+- Argument Annotated Essays Corpus (AAEC) ([Stab and Gurevych, 2017](https://aclanthology.org/J17-3005.pdf)) contains student essays. There are two types of data: essay-level and paragraph-level ([Eger et al., 2017](https://aclanthology.org/P17-1002/)). A stance for a controversial theme is expressed by a `MajorClaim` component as well as `Claim` components, and `Premise` components justify or re ...
```
"There are two types of data: essay-level and paragraph-level" what does this mean? What is essay-level / paragraph-level data?

idalr replied:
elaborated in de8352b.
According to Stab & Gurevych, the diagrams of `Claim` and `Premise` are always completed in a paragraph. Therefore, when training and predicting, it is possible to feed data to a model by paragraph, as well as by essay.

ArneBinder replied:
Ah ok, thanks for the clarification!

ArneBinder commented on AM_documentation/corpora/aae2.md:
```diff
@@ -0,0 +1,235 @@
+| `Claim`      |  1228 |
+| `Premise`    |  3832 |
+
+- `MajorClaim` is the root node of the argumentation structure and represents the author’s standpoint on the topic. Essay bodies either support or attack the
```
How is the `MajorClaim` defined if the current structure has multiple "roots" (above it is written that only 73% of the entries are trees)? Is it each root of the subtrees then?

idalr replied:
elaborated in de8352b.
From what I understand so far, each essay is not a complete tree because they did not explicitly annotate the relation between `Claim` and `MajorClaim` and between claims in the same paragraph. Please see the Example section in the document.

ArneBinder commented on AM_documentation/corpora/argmicro.md:
```diff
@@ -0,0 +1,230 @@
+- reb: rebutting attack (ADU->ADU)
+  - definition: "targeting another node and thereby challenging its acceptability"
+- und: undercutting attack (ADU->Edge)
+  - definition: "targeting an edge and thereby challenging the acceptability of the inference from the source to the target node"
```
The entries of the table do not match the entries of the list (especially `seg` and `sup`, see [here](https://huggingface.co/datasets/DFKI-SLT/argmicro/blob/main/argmicro.py#L37)). What's about that and the missing items them?

idalr replied:
elaborated in de8352b.
There were some changes in statistics that Morio et al. made, including this.
- First, they worked from the ADUs level; therefore, omitting 'seg' and 'joint' with are non-argumentative relations between EDUs and between EDUs and ADUs.
- Second, they removed `add` by merging ADUs that had `add` relations together.

ArneBinder reviewed (changes requested):
We are getting close! This was already a nice read.

Some missing points:
- add new point "pre-defined splits" (e.g. `train`: *num of train docs*, `test`: *num of test docs*; may be just `none` as in the case of sciarg), maybe below original dataset statistic
   - the statistics that you collected should be for these splits 
- add a new point "ADUs contain shell language" (yes or no) (below "segment-based spans", should be the same value as that, but to have it written down explicitly makes it a bit more visible)
- add a new point "text based or token based" (I think only scidtb_argmin is token based, the other datasets should be text based)
- please also add label count statistics for the relations (you can copy the `count_entity_labels.yaml` config and adjust it to collect the labels of the relations instead of the entities)

This is also required, but we should do this in one or multiple follow-up PRs because this is already quite big:
- next to component and relation counts, the percentage would be interesting
- add new point "preprocessing in Morio et al" (how did the authors process the data? e.g. splits, adjusting of components / relations, ...)
- screenshots of examples (such as the pdfs in argmicro)

idalr committed:
updated .md files

Ran pre-commit locally on `base` environment, but still wouldn't allow me to push change to GitHub. This is manual upload.

idalr committed:
test commit

idalr committed:
test commit 2

idalr committed:
delete test.md

idalr committed:
test commit 4

idalr committed:
update data and statistics

ArneBinder committed:
re-arrange content in corpora md files and minor improvements

ArneBinder committed:
fix scidtb-argmin corpus name

ArneBinder committed:
re-add scidtb-argmin md file and move all corpus md files to argumentation-mining/documentation/corpora
